cdap-ui/app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsButtons/PipelineSummaryButton.js

```diff
@@ -77,18 +78,20 @@ export default class PipelineSummaryButton extends Component {
         })}
       >
         {this.renderSummaryButton()}
-        <PipelineSummary
-          pipelineType={pipelineType}
-          namespaceId={getCurrentNamespace()}
-          appId={this.props.pipelineName}
-          programType={programType}
-          programId={programId}
-          pipelineConfig={PipelineDetailStore.getState()}
-          totalRunsCount={PipelineDetailStore.getState().runs.length}
-          onClose={this.toggleSummary.bind(this, false)}
-          anchorEl={this.buttonRef.current}
-          open={this.state.showSummary}
-        />
+        <If condition={this.state.showSummary}>
```

Comment:
Why was it necessary to add `If`? Was `open` in the props not enough?

Reply:
The `PipelineSummary` is loaded with user loads the details page. Once loaded it only listens to prop changes (in our case totalRunsCount).

So without this change, the charts will pick up new runs on each open/close but not the state. So the pipeline run might have completed behind the scenes but the status and runtime won't be updated in the charts as we don't make a new query for runs. 

This change will make sure we query runs for the pipeline everytime the user opens the summary modeless. This wouldn't have been the case if we passed on the runs from angular to react, but that has the downside of diff'ing on each update (which we didn't end up doing)